src/west/manifest.py

```diff
@@ -217,6 +239,26 @@ class MalformedManifest(Exception):
 
 # Definitions for Manifest attribute types.
 
+class WestMeta:
+
+    __slots__ = 'url revision'.split()
+
+    def __init__(self, url=None, revision=None):
+        if url is None:
+            url = WEST_URL_DEFAULT
+        if revision is None:
+            revision = WEST_REV_DEFAULT
+
+        self.url = url
+        self.revision = revision
+
+    def __eq__(self, other):
+        raise NotImplemented
```

Comment:
Side note, but suspect there isn't any harm in allowing comparisons. It uses the identity by the default (`a == b` if `a` and `b` refer to the same object).

Reply:
I also took this from the other classes in `manifest.py`.

Reply:
I wrote those. I thought about allowing equality in each of those particular cases and decided against it.